src/LiquidityPoolConverter.ts

```diff
@@ -204,12 +205,12 @@ class IConversionEvent {
 }
 
 function handleConversion(event: IConversionEvent): void {
-  const fromAmount = decimalize(event.fromAmount, event.fromToken)
-  const toAmount = decimalize(event.toAmount, event.toToken)
-  const conversionFee = decimalize(event.conversionFee, event.toToken)
-  const protocolFee = decimalize(event.protocolFee, event.toToken)
+  const toToken = Token.load(event.toToken.toHexString())
+  const fromAmount = decimal.fromBigInt(event.fromAmount, DEFAULT_DECIMALS)
```

Comment:
Sort of, but I'm using default decimals because:
1. We don't currently have the subgraph set up to handle non-18 decimal tokens
2. The fromToken Token entity isn't loaded already here. It gets loaded at least 3 times throughout handleConversion, which is very slow, so I don't want to load it again. Eventually I want to refactor this so it's only loaded once and then passed to each successive function as a param and only saved at the end

Reply:
> Sort of, but I'm using default decimals because:
> 
>     1. We don't currently have the subgraph set up to handle non-18 decimal tokens
> 
>     2. The fromToken Token entity isn't loaded already here. It gets loaded at least 3 times throughout handleConversion, which is very slow, so I don't want to load it again. Eventually I want to refactor this so it's only loaded once and then passed to each successive function as a param and only saved at the end

1. what does that mean? what is not set up for using non-18 decimal tokens?
2. let's add this refactor as a task on Jira and add it to the backlog

Reply:
Oh actually I think you're right - it is set up to handle non-18 tokens on the AMM, it's just not set up for non-BTC pairs yet.

I've added the refactor task here: https://sovryn.atlassian.net/browse/SOV-685

And I'll change this to use correct decimals